python/activator/middleware_interface.py

```diff
@@ -689,6 +690,48 @@ def _download(self, remote):
         local.transfer_from(remote, "copy")
         return local
 
+    def check_for_snap(self, instrument: str, groupid: str, snap: int,
+                       detector: int) -> tuple[Snap | None, str | None]:
+        """Check to see if an image for a group/snap is already ingested.
+
+        Parameters
+        ----------
+        instrument : `str`
+        groupid : `str`
+        snap : int
+        detector : int
+
+        Returns
+        -------
+        raw_info : `Snap` or None
+            The expanded information about the image, None if it is not
+            present.
+        oid : `str` or None
+            The pathname of the image, None if it is not present.
+        """
+        datasets = set(self.central_butler.registry.queryDatasets(
+            datasetType="raw",
+            collections=f"{instrument}/raw/all",
```

Comment:
I'm no longer doing this — but I'm no longer doing anything.  Since the Kafka consumer is subscribed at the beginning of `next_visit_handler`, I think it's only cold starts that could have an image show up before the subscription.  Otherwise, all image notifications should be picked up by `consumer.consume()`.